Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -0,0 +1,121 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "TRDBase/TrackletTransformer.h"
+#include "TMath.h"
+
+using namespace o2::trd;
+
+TrackletTransformer::TrackletTransformer()
+{
+  mgeo = o2::trd::Geometry::instance();
+  mgeo->createPadPlaneArray();
+}
+
+std::vector<double> TrackletTransformer::transformToSpacePoint(int hcid, int padrow, int column, int position)
+{
+  int detector = hcid / 2;
+  int stack = mgeo->getStack(detector);
+  int layer = mgeo->getLayer(detector);
+  auto padPlane = mgeo->getPadPlane(layer, stack);
+
+  // (I=inner, O=outer) -- inner: pads [1,142] | outer: pads 0,143
+  // pad dimension in rphi-direction
+  double padWidth = padPlane->getWidthIPad();
+
+  // pad dimension in z-direction
+  double padLength = padPlane->getLengthIPad();
+  double oPadLength = padPlane->getLengthOPad();
+
+  int nPadRows = padPlane->getNrows();
+
+  float driftHeight = mgeo->cdrHght();
+
+  // 5mm below cathode plane to reduce error propogation from tracklet fit and driftV
+  double x = driftHeight - 0.5;
```

Comment:
So I think there is still a bit of a discussion around this and the precise position is still not determined. We planned to optimize it later when we had data to test on, but the idea was to report x at some position **below** the cathode plane nonetheless. 

This was initially @tdietel's idea and my understanding is still a bit fuzzy, but maybe this is a good place to talk about it and gain more clarity: My original understanding was along a similar vein as yours where I thought we report the position at the anode plane since any electrons ionized here are absorbed almost instantly and therefore do not have any time to wander around, so we can be fairly sure of at least one point in the detector where the particle went through. Practically, I thought that this would be implemented by taking the pad position of the first cluster. However, I think this actually carries a few additional assumptions which accumulate some uncertainty. In this case, the position is based only on a single data point, the first cluster may not have been ionized directly on the anode plane, other clusters nearby may also be absorbed in the same timebin being examined, and I don't know how easy it is to determine which timebin to look at in the first place. 

So I think the idea is that it is actually more accurate to use the tracklet fit for this information. Now, in this case, the position of the particle at the anode plane has a larger uncertainty than at other positions below the cathode plane. This is caused by two primary sources of uncertainty:

- The first source of uncertainty is from the drift velocity. A change in drift velocity seems to rotate the tracklet about the cathode plane. I am not sure if a change in drift velocity results in any change in the clusters in the amplification region, but even if it did, I believe that the TRAP aims to perform the tracklet fit over the drift region exclusively. This is according to information I found in Jochen Klein's M.Sc. thesis. I actually emailed Venelin about this to confirm and he recommended I consult the TRAP user manual and/or the thesis of Marcus Gutfleisch. I looked at both of these and struggled to find the answer. But I did notice a word in several TRAPconfig names `fs1e24` which makes me think that the fit starts at timebin 1 and ends at timebin 24 in which case maybe it does include some clusters in the amplification region. But I am just speculating at this point...  So this needs to be confirmed. But if this is true, then the point of minimal uncertainty will be on the cathode plane.

- The second source is the tracklet fit itself. This is a straight line fit and as you probably know, any straight line fit has maximal uncertainty at either end and minimal uncertainty at the center of gravity due to the way in which varying the slope and the intercept shift and rotate the line.

Based on these two sources of uncertainty, the minimum uncertainty of the tracklet would be found at a point below the cathode plan and above the tracklet center of gravity. I don't think the tracklet fit is weighted in any way so I suppose the center of gravity would be at the mean timebin used for the fit.

Let me know your thoughts. Would be great to get @tdietel's comments as well. Maybe we can have a meeting sometime to address a few points including this one.

Reply:
The fit is done between TPFS and TPFE, page 56 of the trap user manual, and line 1526 or there abouts in the trapsimulator. There is a method in the current trapconfig to dump it to text. You can download the files as root files off ccdb-test.cern.ch and interrogate or dump it as you see fit.